### PR TITLE
Feat/6686 create microservices event

### PR DESCRIPTION
**Pull request by morhag90**
Close: #6686 

## PR Details

Add new event CreateConnectMicroservicesParams

### Summary
📞🔌📈

This pull request adds a new feature to generate code for connecting microservices based on the application models. It introduces a new plugin event `CreateConnectMicroservices` and a new template file for the connect microservices function. It also updates the data service server and the client to use the new function. Additionally, it adds some SQL scripts to migrate the database fields for decimal and whole numbers.

> _Sing, O Muse, of the mighty code warriors who strove_
> _To connect their app with the cloud services they loved_
> _With the help of `connectMicroservices`, a function divine_
> _Crafted from templates and events by the `@amplication/code-gen-types` shrine_

### Walkthrough
*  Add a function to connect the app to other microservices based on the configuration ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-4c872ee4a5070b372af7443b69fa0846d5f4697115a7d94ea3dd789a36f8c121R1-R6), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5fb3e22c3521656b07427461762e402241287c0c56ce20e348c545e75b291dbeR1-R37), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR25), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR106-R108), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR130), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-880f68212611e24dc180018dd0af420b064367cea7b3d897405eede2ba80f315R8-R10), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-880f68212611e24dc180018dd0af420b064367cea7b3d897405eede2ba80f315R46-R48))
  - Import the `ConfigService` from NestJS and define a function that takes an `INestApplication` as a parameter in the template file `connect-microservices.template.ts` ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-4c872ee4a5070b372af7443b69fa0846d5f4697115a7d94ea3dd789a36f8c121R1-R6))
  - Implement the logic for creating the connect microservices function in the file `connect-microservices.ts` using the types and events from the `@amplication/code-gen-types` package, the `DsgContext` from the `data-service-generator` package, the `pluginWrapper` utility, and the `readFile` and `print` utilities ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5fb3e22c3521656b07427461762e402241287c0c56ce20e348c545e75b291dbeR1-R37))
  - Import the `connectMicroservices` function and call it inside the `createServerInternal` function in the `create-server.ts` file, and add the result to the array of modules that are passed to the `writeModules` function ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR25), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR106-R108), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR130))
  - Import the `connectMicroservices` function and call it twice inside the `main` function in the `main.ts` file in the `static/src` folder, and also call the `app.startAllMicroservices` function ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-880f68212611e24dc180018dd0af420b064367cea7b3d897405eede2ba80f315R8-R10), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-880f68212611e24dc180018dd0af420b064367cea7b3d897405eede2ba80f315R46-R48))
* Update the `@amplication/code-gen-types` package to support the connect microservices feature ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-848fe18cdea81eff2bb3d7e509a3d8e5a63a2974014e58ee366093f9fa9f2bf9L3-R3), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-8ec1349510d73b28d516206efd7feaeaade54b394a928088f17e945bd9312ce3R180-R183), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R37), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R65), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-3811ed265e535dd8e3e3bd8618ca04a8bd0860e1cd5f0bd9e9cf2e8d321f321aR109))
  - Bump the version of the package from 2.0.4 to 2.0.5 in the `package.json` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-848fe18cdea81eff2bb3d7e509a3d8e5a63a2974014e58ee366093f9fa9f2bf9L3-R3))
  - Add a new interface `CreateConnectMicroservicesParams` that defines the parameters for the event `CreateConnectMicroservices` in the `plugin-events-params.ts` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-8ec1349510d73b28d516206efd7feaeaade54b394a928088f17e945bd9312ce3R180-R183))
  - Import the `CreateConnectMicroservicesParams` interface and add a new property to the `Events` type that maps the event name `CreateConnectMicroservices` to the plugin event type in the `plugin-events.ts` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R37), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R65))
  - Add a new enum value `CreateConnectMicroservices` to the `EventNames` enum in the `plugins-types.ts` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-3811ed265e535dd8e3e3bd8618ca04a8bd0860e1cd5f0bd9e9cf2e8d321f321aR109))
* Add SQL scripts to update the `databaseFieldType` property for fields with the `dataType` of `DecimalNumber` or `WholeNumber` in the `EntityField` table ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-fef4b4f90e55625cd0e7d233efafd9f8a8039ad21ea60b1223441d34f777a5d6L1-R8), [link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-21895e8d15c90b7c55b0330c8f9b361b16cbc0a503be834e188c1beaa3172990R1-R10))
  - Add a script that sets the `databaseFieldType` property to `FLOAT` for fields with the `dataType` of `DecimalNumber` and no existing `databaseFieldType` in the `20230809_add_decimal_datatype.sql` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-fef4b4f90e55625cd0e7d233efafd9f8a8039ad21ea60b1223441d34f777a5d6L1-R8))
  - Add a script that sets the `databaseFieldType` property to `INT` for fields with the `dataType` of `WholeNumber` and no existing `databaseFieldType` in the `20230809_add_wholenumber_datatype.sql` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-21895e8d15c90b7c55b0330c8f9b361b16cbc0a503be834e188c1beaa3172990R1-R10))
* Add a function to open a contact us page when the user clicks on the sync new git organization button for AWS CodeCommit in the `GitProviderConnectionList.tsx` file ([link](https://github.com/amplication/amplication/pull/6692/files?diff=unified&w=0#diff-5af106441ba9187b6775a40d93abe24a8d709f56acd1bb7a60d58a1167fe4ef8L95-R97))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-code-gen-types/src/plugin-events.ts [hunk 1]
```diff
@@ -61,6 +62,7 @@ export type Events = {
   [EventNames.CreateServerPackageJson]?: PluginEventType<CreateServerPackageJsonParams>;
   [EventNames.CreateAdminUIPackageJson]?: PluginEventType<CreateAdminUIPackageJsonParams>;
   [EventNames.CreateServerAppModule]?: PluginEventType<CreateServerAppModuleParams>;
+  [EventNames.CreateConnectMicroservices]?: PluginEventType<CreateConnectMicroservicesParams>;
```
Comment (overbit) [inline packages/amplication-code-gen-types/src/plugin-events.ts:65]: Let's make an update to the docs with an example of how to add a microservice. :)

**Review by overbit — APPROVED** [review]

**Review by mulygottlieb — APPROVED** [review]

File: packages/data-service-generator/src/server/static/src/main.ts [hunk 2]
```diff
@@ -5,6 +5,9 @@ import { HttpExceptionFilter } from "./filters/HttpExceptions.filter";
 // @ts-ignore
 // eslint-disable-next-line
 import { AppModule } from "./app.module";
+// @ts-ignore
+// eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (mulygottlieb) [reply packages/data-service-generator/src/server/static/src/main.ts:9]: @overbit I am resolving this - please open a separate ticket to review @ts-ignore and eslint-disable in the generated code

[the remark this answers is not in the document]
Comment (mulygottlieb) [reply packages/data-service-generator/src/server/static/src/main.ts:9]: https://github.com/amplication/amplication/issues/6703

File: packages/amplication-code-gen-types/src/plugin-events.ts (hunk 1 above)
Comment (mulygottlieb) [reply packages/amplication-code-gen-types/src/plugin-events.ts:65]: https://github.com/amplication/amplication/issues/6704

**Pull request merged**